indico/modules/events/registration/fields/simple.py

```diff
@@ -89,32 +88,42 @@ def filter_choices(self):
 
     @classmethod
     def process_field_data(cls, data, old_data=None, old_versioned_data=None):
-        unversioned_data, versioned_data = super(SingleChoiceField, cls).process_field_data(data, old_data,
-                                                                                            old_versioned_data)
+        unversioned_data, versioned_data = super(ChoiceBaseField, cls).process_field_data(data, old_data,
+                                                                                          old_versioned_data)
         items = [x for x in versioned_data['choices'] if not x.get('remove')]
         captions = dict(old_data['captions']) if old_data is not None else {}
-        default_item = None
+        if cls.has_default_item:
+            unversioned_data.setdefault('default_item', None)
         for item in items:
             if 'id' not in item:
                 item['id'] = unicode(uuid4())
             item.setdefault('is_billable', False)
-            item['price'] = float(item['price']) if item['price'] else 0
+            item['price'] = float(item['price']) if item.get('price') else 0
             item['places_limit'] = int(item['places_limit']) if item.get('places_limit') else 0
-            # XXX: it would be nice if we could use item['is_default'] but doing that with angular is tricky
-            if unversioned_data.get('default_item') in {item['caption'], item['id']}:
-                default_item = item['id']
+            item['max_extra_slots'] = int(item['max_extra_slots']) if item.get('max_extra_slots') else 0
+            if cls.has_default_item and unversioned_data.get('default_item') in {item['caption'], item['id']}:
```

Comment:
since we setdefault the default item you can use `unversioned_data['default_item']` here